apisix/cli/ngx_tpl.lua

```diff
@@ -459,11 +459,11 @@ http {
 
     server {
         {% for _, item in ipairs(node_listen) do %}
-        listen {* item.port *} default_server {% if enable_reuseport then %} reuseport {% end %} {% if item.enable_http2 then %} http2 {% end %};
-        {% end %}
-        {% if ssl.enable then %}
+        listen {* item.ip *}:{* item.port *} default_server {% if enable_reuseport then %} reuseport {% end %} {% if item.enable_http2 then %} http2 {% end %};
+        {% if item.allow_ssl and ssl.enable then %}
```

Comment:
Better to use enable_ssl.

Reply:
ok，and should I explain these params in the doc FAQ.md ?
```yaml
apisix:
  node_listen:
  - 9080
  - port: 9380
    ip: 127.0.0.1            # If not given, the default is `0.0.0.0`.
    enable_ssl: True         # If `False`, the ip connot be ued to enable ssl. If not given, the default is `True`.
```

Reply:
IMHO, document it in the conf/config-default.yaml as a comment is enough.

Reply:
And the default value of "enable_ssl" should not be true, since `9080` doesn't have "enable_ssl", and it doesn't enable SSL by default.

Reply:
Actually, we are planning to reduce the length of the current FAQ, let each answer go to suitable places, instead of putting them together.

Reply:
> And the default value of "enable_ssl" should not be true, since `9080` doesn't have "enable_ssl", and it doesn't enable SSL by default.

If  the default value of "enable_ssl" is false, the old configuration will be affected.
Considering the param `enable_ssl` should be compatible with the original param `apisix.ssl.enable`, I design the param `enable_ssl`  as a control switch that cooperates with the param  `apisix.ssl.enable`:
```lua
{% if item.enable_ssl and ssl.enable then %}
{% for _, port in ipairs(ssl.listen_port) do %}
listen {* item.ip *}:{* port *} ssl default_server {% if ssl.enable_http2 then %} http2 {% end %} {% if enable_reuseport then %} reuseport {% end %};
{% end %}
{% end %}
```
Beacause there are other params about ssl ( `enable_http2`, `listen_port` etc. ) is set in the param `apisix.ssl`, the ssl config should not be complex in the param `apisix.node_listen[]`
```yaml
apisix
  ssl:
    enable: true
    enable_http2: true
    listen_port: 9443
```

Reply:
It is better to modify the config `apisix.ssl.listen_port` ,like this:
```yaml
apisix:
  ssl:
    listen_port:
    - ip: 127.0.0.1
       port: 9443
```
The ssl config in the param `apisix.node_listen[]` could be confusing，for example:
```yaml
apisix:
  node_listen:
    - ip: 127.0.0.2
      port: 9082
      enable_ssl: true
    - ip: 127.0.0.2
      port: 9083
      enable_ssl: false
```
What do you think?  @spacewander

Reply:
Agree. Personally, I prefer to use a new key "listen" since it is not just with port. Maybe we need to move "enable_http2" under "listen" too. Like this one:
```
apisix:
  ssl:
    # enable_http2: false
    # listen_port: ...
    listen:
    - ip: 127.0.0.1
      port: 9443
      enable_http2: true # per listen configuration
```

@wayne-cheng
What's your opinion?

Reply:
Yes. We need to keep backward compatibility.

Reply:
@spacewander I have updated the code, please take a look.